code/modules/projectiles/projectile/bullets/shotgun.dm

```diff
@@ -124,6 +124,47 @@
 	stamina = 13 //Total of 78 with less falloff (very big)
 	sharpness = SHARP_NONE
 
+// Normal airburst
+
+/obj/item/projectile/bullet/pellet/shotgun_airburst_casing
+	name = "airburst rubber casing"
+	damage = 2 // The shell for the bullets, not designed to hit people
+	range = 7 // roughly 4 tiles 
+	sharpness = SHARP_NONE
+
+/obj/item/projectile/bullet/pellet/shotgun_airburst_casing/on_range()
+	var/obj/item/ammo_casing/shotgun/airburst_pellet_shell/P = new(get_turf(src))
+	var/mob/living/L = new (get_turf(src))//it's jank, but casings can only be shot via a mob's location
```

Comment:
WTF...?
You should make ``fire_casing`` pass by when the user arg isn't passed, not make a whole new person for every single bullet!

If you want to go the extra mile, you could even make the user arg be 'source of the firing', and check if it's a mob when necessary, so you can have the 'source' be this pellet instead.